### ISSUE TITLE
Error on Julia v 0.6

### ISSUE THREAD

**Issue by mirestrepo** [cstjean/ScikitLearn.jl#30]
After upgrading to Julia v0.6 
ERROR: LoadError: Module ScikitLearn declares __precompile__(true) but require failed to create a usable precompiled cache file.
Stacktrace:

running Pkg.build("ScikitLearn") after fixes it but should that be necessary?

**Comment by cstjean** [comment]
Yeah, I've hit that a few times during the 0.6 alpha period, but restarting Julia usually solved the problem. I've never used `Pkg.build("ScikitLearn")`. I [posted a question about it on Discourse](https://discourse.julialang.org/t/require-failed-to-create-a-usable-precompiled-cached-file/4548/1), we'll see if there's a solution. Thank you for the bug report.

**Comment by stevengj** [comment]
Possibly related to JuliaLang/julia#18150 and JuliaLang/julia#12508 ... precompiling succeeds, but it can't be loaded because you already have an older conflicting version of a module loaded.  Nothing to do but to restart Julia.

**Comment by cstjean** [comment]
Ok, thank you for looking into it.

**Comment by dsweber2** [comment]
I've been having this issue with 0.6.2. Restarting did make the problem go away. However, when I run from within Atom, the problem refuses to go away, even after restarting atom. I'm guessing there are some strange interactions going on with https://github.com/JunoLab/Atom.jl

**Comment by harryscholes** [comment]
Same. In Atom I'm getting error:

```
julia> using ScikitLearn
INFO: Recompiling stale cache file /Users/harry/.julia/lib/v0.6/ScikitLearn.ji for module ScikitLearn.
WARNING: The call to compilecache failed to create a usable precompiled cache file for module ScikitLearn. Got:
WARNING: Module Iterators uuid did not match cache file.
```

But import works fine in the REPL.

**Comment by fdgodabhi** [comment]
Same is the case with my system. Error when using in Atom but not otherwise.

**Comment by stevengj** [comment]
Probably you should file an issue with JunoLab/atom-julia-client … I'm guessing there is some issue with the Julia process persisting even when you quit Atom?

**Comment by cstjean** [comment]
I'm pretty sure that these problems are caused by the hokey reexporting I do in the submodules. I just don't have time to fix it at the moment (would welcome a PR if anyone's interested, but it's not trivial).

Does anyone object if I turn off precompilation in the meanwhile? It seems to affect many users.
